### PR TITLE
DOC-127: Make Content Encryption Algorithm Configurable

### PR DESCRIPTION
**Pull request by yatsenko-ihor**

### CODE REVIEW

File: datasafe-business-api/src/main/java/de/adorsys/datasafe/business/api/encryption/cmsencryption/CMSEncryptionService.java [hunk 1]
```diff
@@ -8,12 +8,15 @@
 
 import de.adorsys.datasafe.business.api.deployment.keystore.types.KeyID;
 import de.adorsys.datasafe.business.api.deployment.keystore.types.KeyStoreAccess;
+import org.bouncycastle.asn1.ASN1ObjectIdentifier;
```
Comment (valb3r) [inline datasafe-business-api/src/main/java/de/adorsys/datasafe/business/api/encryption/cmsencryption/CMSEncryptionService.java]: I thin we can inject algorithm configuration using dagger - i.e. create some EncryptionConfig interface with default implementation and user will decide what to use

Comment (pn-iyatsenko) [reply datasafe-business-api/src/main/java/de/adorsys/datasafe/business/api/encryption/cmsencryption/CMSEncryptionService.java]: Ok, good idea. What do you think, just an algorithm filed is enough for this config, or something other should be also in config?

Comment (valb3r) [reply datasafe-business-api/src/main/java/de/adorsys/datasafe/business/api/encryption/cmsencryption/CMSEncryptionService.java]: I think we can start with something simple like just algorithm field

File: datasafe-business-api/src/main/java/de/adorsys/datasafe/business/api/types/CMSEncryptionConfig.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package de.adorsys.datasafe.business.api.types;
+
+import lombok.AllArgsConstructor;
+import org.bouncycastle.asn1.ASN1ObjectIdentifier;
+
+@AllArgsConstructor
+public class CMSEncryptionConfig {
```
Comment (valb3r) [inline datasafe-business-api/src/main/java/de/adorsys/datasafe/business/api/types/CMSEncryptionConfig.java]: Why we need concrete class and not an interface here ?

**Pull request merged**